js/widgets/phrasemaker.js

```diff
@@ -488,57 +485,57 @@ function PitchTimeMatrix() {
                     let eCell = event.target;
                     if (eCell.getAttribute("alt") === null) {
                         eCell = eCell.parentNode;
                     }
-                    let index = eCell.getAttribute("alt").split("__")[0];
-                    let condition = eCell.getAttribute("alt").split("__")[1];
-                    that._createColumnPieSubmenu(index, condition);
+                    const index = eCell.getAttribute("alt").split("__")[0];
+                    const condition = eCell.getAttribute("alt").split("__")[1];
+                    this._createColumnPieSubmenu(index, condition);
                 };
 
                 this._noteStored.push(drumName);
             } else if (this.rowLabels[i].slice(0, 4) === "http") {
                 cell.innerHTML = this.rowLabels[i];
                 cell.style.fontSize = Math.floor(this._cellScale * 14) + "px";
                 this._noteStored.push(this.rowLabels[i].replace(/ /g, ": "));
-            } else if (MATRIXSYNTHS.indexOf(this.rowLabels[i]) !== -1) {
+            } else if (PhraseMaker.MATRIXSYNTHS.indexOf(this.rowLabels[i]) !== -1) {
                 cell.innerHTML = this.rowArgs[i];
                 cell.style.fontSize = Math.floor(this._cellScale * 14) + "px";
                 cell.setAttribute("alt", i + "__" + "synthsblocks");
 
-                cell.onclick = function(event) {
+                cell.onclick = (event) => {
                     let eCell = event.target;
                     if (eCell.getAttribute("alt") === null) {
                         eCell = eCell.parentNode;
                     }
-                    let index = eCell.getAttribute("alt").split("__")[0];
-                    let condition = eCell.getAttribute("alt").split("__")[1];
-                    that._createMatrixGraphicsPieSubmenu(
+                    const index = eCell.getAttribute("alt").split("__")[0];
+                    const condition = eCell.getAttribute("alt").split("__")[1];
+                    this._createMatrixGraphicsPieSubmenu(
                         index,
                         condition,
                         null
                     );
                 };
 
                 this._noteStored.push(this.rowArgs[i]);
-            } else if (MATRIXGRAPHICS.indexOf(this.rowLabels[i]) !== -1) {
-                blockLabel = this._logo.blocks.protoBlockDict[
+            } else if (PhraseMaker.MATRIXGRAPHICS.indexOf(this.rowLabels[i]) !== -1) {
+                blockLabel = logo.blocks.protoBlockDict[
                     this.rowLabels[i]
                 ]["staticLabels"][0];
                 cell.innerHTML = blockLabel + "<br>" + this.rowArgs[i];
                 cell.style.fontSize = Math.floor(this._cellScale * 12) + "px";
                 cell.setAttribute("alt", i + "__" + "graphicsblocks");
 
-                cell.onclick = function(event) {
+                cell.onclick = (event) => {
                     eCell = event.target;
```

Comment:
`let eCell`

Reply:
Get this fixed.

Reply:
Fixed this and all similar issues. Also enabled `no-undef` in `eslint`, added a list of global vars with where they come from the head of the file, it might prove to be useful when we start using es6 modules. Thanks